vroom/result.py

```diff
@@ -0,0 +1,31 @@
+"""Result type."""
+
+from collections import namedtuple
+from enum import Enum
+
+class ResultType(Enum):
+  result = 1
+  error = 2
+
+# Inherit from namedtuple so we get an immutable value.
+class Result(namedtuple('Result', ['status', 'value'])):
+  """Holds the result or error of a function call.
+
+  status should be one of ResultType.result or ResultType.error
+  """
+
+  @classmethod
+  def Result(cls, value):
```

Comment:
Can we just have a result.Success or something semantic like that? AFAICT you never actually read the True values and True is just a placeholder.